File: LeetCode/twoSums/twoSums.py
from curses.ascii import SO


class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        valuesMap = {}
        for i in range(len(nums)):
            valuesMap[nums[i]] = i
        # print(valuesMap)
        for i in range(len(nums)):
            lookupTarget = target - nums[i]
            valueFromMap = valuesMap.get(lookupTarget)
            if (valueFromMap):
                if (valuesMap[lookupTarget] != i):
                    return [i, valuesMap[lookupTarget]]


calculator = Solution()

print(calculator.twoSum([3, 2, 4], 6))
print(calculator.twoSum([3, 3], 6))

File: LeetCode/twoSums/twoSums.js
// https://leetcode.com/problems/two-sum/

var twoSum = function (nums, target) {
  const numObj = {};

  for (let i = 0; i < nums.length; i++) {
    numObj[nums[i]] = i;
  }

  for (let i = 0; i < nums.length; i++) {
    const lookupValue = target - nums[i];
    if (numObj[lookupValue] && i !== numObj[lookupValue]) {
      return [i, numObj[lookupValue]];
    }
  }
};

console.log(twoSum([3, 2, 4], 6));
console.log(twoSum([3, 3], 6));
